abstract/group.go

```diff
@@ -49,6 +49,9 @@ type Scalar interface {
 
 	// Set to the modular inverse of scalar a
 	Inv(a Scalar) Scalar
+	
+	// Set to the gcd of a and its module
+	Gcd(a Scalar) Scalar
```

Comment:
Have you seen the inv method?I haven't written it (it existed before I wrote the gcd()) and it receives only one parameter, so I thought it was ok to add another method which receives only one parameter.  I can change it to receive two arguments. Anyway, I dont think that the second option is very generic so it is better to leave it in abstract as it is now. Regarding x and y-I dont need them for my protocol, so I didn't included them so I think I will leave it as it is unless you think differently.